### PR TITLE
Feature/remove re reducers

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/AccountInfoService.java [hunk 1]
```diff
@@ -147,14 +205,18 @@ private JSONObject getOwnBalance() {
 
 		return jsonObject()
 			.put("tokens", balancesArray)
+			.put("prepared_stakes", preparedStakesArray)
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/AccountInfoService.java:208]: Please, rename field to camel case for consistency

File: radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/SystemMapKey.java [hunk 2]
```diff
@@ -47,6 +47,13 @@ public static SystemMapKey ofValidatorData(byte typeId, byte[] mapKey) {
 		return new SystemMapKey(buf.array());
 	}
 
+	public static SystemMapKey ofResourceData(REAddr addr, byte typeId) {
```
Comment (siy) [inline radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/SystemMapKey.java:50]: Argument order looks inconsistent with other methods

**Review by siy — CHANGES_REQUESTED** [review]

File: radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/SystemMapKey.java (hunk 2 above)
Comment (iamyulong) [reply radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/SystemMapKey.java:50]: +1 on consistency.

Currently, we have:
- `0x00 + (epoch_data | round_data | unclaimed_address)  =>  substate_id`: system data
- `0x02 + (validator_rake | validator_owner | validator_metadata | validator_allow_delegation)  =>  substate_id`: virtual parents
- `0x02 + (validator_rake | validator_owner | validator_metadata | validator_allow_delegation) + validator_public_key  =>  substate_id`: validator data
- `0x03 + hashed_key + (resource | resource_metadata)  =>  substate_id`: resource data

Suggest to have `address + substate_type_id + bytes`, assuming 
- `0x02+public_key` is validator address (internal)
- The validator data virtual parents are of system

In short, we have:
```
0x00 + <epoch_data | round_data | unclaimed_address | validator_xxx>  => substate id
0x02 + validator_public_key + <validator_xxx>  => substate id
0x03 + hashed_key + <resource_xxx>  => substate id
```

**Review by iamyulong — APPROVED** [review]
LGTM
